### PR TITLE
[WIP] Add edit and delete

### PR DESCRIPTION
**Pull request by HansGabriel**

### CODE REVIEW

File: screens/Transactions/Transactions.tsx [hunk 1]
```diff
@@ -115,7 +122,10 @@ export default function Transactions({ navigation }: RootNavigationProps) {
           <Text className="text-gray-500">₱{formatNumber(price)}</Text>
         </View>
         {renderChat}
-        <TouchableOpacity className="absolute bottom-5 right-5 rounded-full bg-[#FFD700] p-[10px]">
+        <TouchableOpacity
+          className="absolute bottom-5 right-5 rounded-full bg-[#FFD700] p-[10px]"
+          onPress={goToEditSell(product)}
```
Comment (ezerssss) [inline screens/Transactions/Transactions.tsx:127]: `() => goToEditSell(product)`

File: screens/SpecificProduct/SpecificProduct.tsx [hunk 2]
```diff
@@ -103,7 +111,10 @@ function SpecificProduct({ route, navigation }: ProductNavigationProps) {
   }
 
   const renderEditButton = user?.displayName === product.seller && (
-    <TouchableOpacity className="absolute right-7 top-14 z-20 h-12 w-12 items-center justify-center rounded-full bg-secondary-100">
+    <TouchableOpacity
+      className="absolute right-7 top-14 z-20 h-12 w-12 items-center justify-center rounded-full bg-secondary-100"
+      onPress={goToEditSell(product)}
```
Comment (ezerssss) [inline screens/SpecificProduct/SpecificProduct.tsx:116]: `() => goToEditSell(product)`

File: helpers/upload.ts [hunk 3]
```diff
@@ -36,9 +35,10 @@ export async function compressImage(uri: string): Promise<string> {
...
 export function getImageID(url: string): string {
-  const { pathname } = new URL(url);
+  const startIdx = url.lastIndexOf('%2F') + 3;
+  const endIdx = url.indexOf('?');
 
-  const productSplice = pathname.split(productURLSplicer);
+  const specificPart = url.substring(startIdx, endIdx);
 
-  return productSplice[1];
+  return specificPart;
```
Comment (ezerssss) [inline helpers/upload.ts:43]: is this more robust than my original implementation? if so, how so?

File: screens/Sell/EditSell.tsx
Comment (ezerssss) [inline screens/Sell/EditSell.tsx:1]: this seems to be the exact same screen as `Sell.tsx` with just additional buttons for `Update` and some additional functions, can you instead:

have an optional screen parameter sa `Sell.tsx`, and you can then set the states with the parameter but you just need to check if the parameter exist, something like:

`const [title, setTitle] = useState(product.title ?? "");`

you can then just conditionally render the update button if the parameter exist

**Review by ezerssss — CHANGES_REQUESTED** [review]

File: helpers/upload.ts (hunk 3 above)
Comment (HansGabriel) [reply helpers/upload.ts:43]: the original implementation wasn't working somehow. perhaps there's a library sa firebase to get the file name only.

Comment (ezerssss) [reply helpers/upload.ts:43]: image

asked manong chatGPT and we kinda have the same implementation

BUT anyways

does this work na? i was also thinking maybe sa reference sa function mismo ang wrong?

File: services/product.ts [hunk 4]
```diff
@@ -47,15 +47,17 @@ export async function uploadProductPhotos(
...
 
-async function deleteProductPhotos(product: Product) {
+export async function deleteProductPhotos(product: Product) {
   try {
     const { images } = product;
 
     for (let i = 1; i < 3; i++) {
       const imageID = getImageID(images[i]);
-      const imageRef = ref(storage, `products/${imageID}`);
+      const imagePath = `products/${imageID}`;
```
Comment (ezerssss) [inline services/product.ts:56]: this is what i mean na wrong reference, maybe we should a `.jpeg` sa imageID?

we can add it here like 
```
const imagePath = `products/${imageID}.jpeg`;
````

or I think even better is sa `getImageID` returned na there ang `.jpeg`

File: helpers/upload.ts (hunk 3 above)
Comment (ezerssss) [reply helpers/upload.ts:43]: maybe add something like
```
return `${specificPart}.jpeg`
```

Comment (HansGabriel) [reply helpers/upload.ts:43]: yeah forgot the jpeg haha

File: screens/Sell/EditSell.tsx
Comment (HansGabriel) [reply screens/Sell/EditSell.tsx:1]: Um I initially planned to reuse the component but I feel like the logic would be too coupled. Usually I try to follow the rule of thirds where if it's repeated the third time, then we'll refactor it. What do you think?

https://en.wikipedia.org/wiki/Rule_of_thirds

Comment (ezerssss) [reply screens/Sell/EditSell.tsx:1]: ooohh okok, TIL haha thanks!

yeah we can keep it this way

**Review by ezerssss — APPROVED** [review]

**Pull request merged**